diff --git a/.pio/libdeps/$$$env$$$/ESPAsyncWebServer/src/WebHandlers.cpp b/.pio/libdeps/$$$env$$$/ESPAsyncWebServer/src/WebHandlers.cpp
index 44a4601..06991ba 100644
--- a/.pio/libdeps/$$$env$$$/ESPAsyncWebServer/src/WebHandlers.cpp
+++ b/.pio/libdeps/$$$env$$$/ESPAsyncWebServer/src/WebHandlers.cpp
@@ -224,6 +224,10 @@ void AsyncStaticWebHandler::handleRequest(AsyncWebServerRequest* request) {
     } else {
       etag = request->_tempFile.size();
     }
+    // As per RFC7232 ( https://www.rfc-editor.org/rfc/rfc7232#section-2.3 )
+    // and MDN docs ( https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/ETag#directives )
+    // the value of the ETag in the http header response has to be between quotes.
+    etag = "\"" + etag + "\"";

     bool not_modified = false;

#
